magefile.go

```diff
@@ -8,31 +8,159 @@
+	"github.com/elastic/elastic-agent-shipper/tools"
+	"github.com/pkg/errors"
+
+	//mage:import
+
 	"github.com/elastic/elastic-agent-libs/dev-tools/mage/gotool"
 	"github.com/magefile/mage/mg"
 	"github.com/magefile/mage/sh"
 )
 
-var (
-	goLicenserRepo = "github.com/elastic/go-licenser@v0.4.1"
+const (
+	GoreleaserRepo = "github.com/goreleaser/goreleaser@v1.6.3"
 )
 
 // Aliases are shortcuts to long target names.
 // nolint: deadcode // it's used by `mage`.
 var Aliases = map[string]interface{}{
-	"llc":  mage.Linter.LastChange,
-	"lint": mage.Linter.All,
+	"build":           Build.Binary,
+	"unitTest":        Test.Unit,
+	"integrationTest": Test.Integration,
+}
+
+// BUILD
+
+// Build contains targets related to linting the Go code
+type Build mg.Namespace
+
+// All builds binaries for the all os/arch.
+func (Build) All() {
+	os.Setenv("PLATFORM", "all")
+	mg.Deps(Build.Binary)
+}
+
+// Clean removes the build directory.
+func (Build) Clean(test string) {
+	os.RemoveAll("build") // nolint:errcheck //not required
 }
 
-func Build() {
-	sh.Run("go", "build")
+// CheckBinaries checks if the binaries are generated (for now).
+func (Build) CheckBinaries() error {
+	path := filepath.Join("build", "binaries")
+	for _, platform := range devtools.PlatformFiles {
+		var execName = devtools.ProjectName
+		if strings.Contains(platform, "windows") {
+			execName += ".exe"
+		}
+		binary := filepath.Join(path, fmt.Sprintf("%s-%s-%s", devtools.ProjectName, tools.DefaultBeatVersion, platform), execName)
+		if _, err := os.Stat(binary); err != nil {
+			return errors.Wrap(err, "Build: binary check failed")
+		}
+	}
+
+	return nil
+}
+
+// InstallGoReleaser target installs goreleaser
+func InstallGoReleaser() error {
+	return gotool.Install(
+		gotool.Install.Package(GoreleaserRepo),
+	)
 }
 
-// Check runs all the checks
+// Binary will create the project binaries found in /build/binaries (use `mage build`)
+// ENV PLATFORM = all, darwin, linux, windows, darwin/amd64, darwin/arm64, linux/386, linux/amd64, linux/arm64, windows/386, windows/amd64
+// ENV SNAPSHOT = true/false
+// ENV DEV = true/false
+func (Build) Binary() error {
+	InstallGoReleaser()
+
+	args := []string{"build", "--rm-dist", "--skip-validate", "--debug"}
+
+	// Environment variable
+	env := map[string]string{
+		"CGO_ENABLED":     devtools.EnvOrDefault("CGO_ENABLED", "0"),
+		"DEV":             devtools.EnvOrDefault("DEV", "false"),
+		"DEFAULT_VERSION": tools.DefaultBeatVersion,
+	}
+
+	versionQualifier, versionQualified := os.LookupEnv("VERSION_QUALIFIER")
+	if versionQualified {
+		env["VERSION_QUALIFIER"] = versionQualifier
+		env["DEFAULT_VERSION"] += fmt.Sprintf("-%s", versionQualifier)
+	}
+
+	if snapshotEnv := os.Getenv("SNAPSHOT"); snapshotEnv != "" {
+		isSnapshot, err := strconv.ParseBool(snapshotEnv)
+		if err != nil {
+			return err
+		}
+		if isSnapshot {
+			args = append(args, "--snapshot")
+			env["DEFAULT_VERSION"] += fmt.Sprintf("-%s", "SNAPSHOT")
```

Comment:
[Suggestion]
the `fmt.Sprintf` is unnecessary
```suggestion
			env["DEFAULT_VERSION"] += "-SNAPSHOT"
```